File: services/usersPy/get.py
from libs.responseLib import success, failure;
from libs.dataExtraction import extractUserSubFromEvent;
import boto3;
from os import environ;
dynamodb = boto3.resource('dynamodb', region_name=environ.get('REGION'));
table = dynamodb.Table(environ.get('tableName'));

def handler(event, context):
    try:
        _event = json.load(event)
    except Exception as e:
        _event = event
    userId = _event['pathParameters']['userId'] if _event['pathParameters'] and _event['pathParameters']['userId'] else extractUserSubFromEvent(_event)
    try:
        key = {
            'pK': f"USER#{userId}",
            'sK': "USER"
        }
        userQuery = table.get_item(
            Key=key
        )
        return success(userQuery['Item']);
    except Exception as e:
        return failure({'message': str(e)});

File: services/userJS/get.js
import { success, failure } from "../../libs/responseLib";
import {extractIdentityIdFromEvent} from "../../libs/dataExtraction";
import dynamoDb from "../../libs/dynamodb-lib";

export async function handler(event, context) {
  try {
    const params = {
      TableName: process.env.tableName,
        Key: {
          pK: `USER#${event.pathParameters.userId || extractIdentityIdFromEvent(event)}`,
          sK: `USER`
        }
    };
    const userQuery = await dynamoDb.get(params);
    return success(userQuery.Item);
  } catch (e) {
    return failure({
      message: "Error reading user",
      error: e.message
    });
  }
}
